src/Workspaces/Core/Desktop/Workspace/Host/TemporaryStorage/TemporaryStorageServiceFactory.MemoryMappedInfo.cs

```diff
@@ -96,27 +104,22 @@ private void ForceCompactingGC()
             /// </summary>
             public Stream CreateReadableStream()
             {
-                // CreateViewStream is not guaranteed to be thread-safe
-                lock (_memoryMappedFile)
+                // CreateViewAccessor is not guaranteed to be thread-safe
+                lock (_memoryMappedFile.Target)
```

Comment:
Just want to confirm: this is safe?  i.e. as long as we ahven't disposed _memoryMappedFile ourselves, then .Target will be non null.  And, if we had disposed, then this would throw... but tha'ts ok because it's not safe to use this after disposing.  Right?